streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/SubmitApplicationRequest.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.streaming.appmaster
+
+import org.apache.gearpump.streaming._
+import org.apache.gearpump.util.{Graph, LogUtil}
+import org.slf4j.Logger
+import spray.http.HttpRequest
+import spray.httpx.unmarshalling.Deserializer
+import upickle.{Js, Reader}
+
+import scala.util.{Failure, Success, Try}
+
+case class SubmitApplicationRequest (
+    appName: String = null,
+    appJar: String = null,
+    processors: Map[ProcessorId, ProcessorDescription] = null,
+    dag: Graph[Int, String] = null)
+
+object SubmitApplicationRequest {
+  private val LOG: Logger = LogUtil.getLogger(getClass)
```

Comment:
This is very similar to object StreamingAppMasterDataDetail which does a 'custom' upickle.Writer. In this case we do a custom upickle.Reader. The complication is how we serialize Graph which is custom. I if we just do a custom read, write on Graph then both StreamingAppMasterDataDetail and SubmitApplicationRequest could use the default read, write. I'll open a separate task on this. We also need to have the same scala code appear on the client using scala.js